### ISSUE TITLE
[Feature Request] Output JSON

### ISSUE THREAD

**Issue by jspeed-meyers** [interlynk-io/sbomqs#38]
Great tool!

An ability via a command-line flag to output the results in JSON would be much appreciated. I know I would benefit. Perhaps other users would benefit too.

**Comment by riteshnoronha** [comment]
Thanks for your feedback @jspeed-meyers  we are working on this, should be in next weeks release.

**Comment by riteshnoronha** [comment]
Here is the sample output of the json, we plan to support.  Feedback welcome.

@surendrapathak @kchetans please review based on our discussions. 

_run_id:_ provides a UUID for the run that generated this output
_timestamp:_ provides the run timestamp, when this report was generated. In UTC
_files: id:_ uniq id created by hashing primary_product, version & download_location 

```json
{
  "run_id": "5c7b85b1-e5ac-41f2-94f9-2b2e8c9ff79a",
  "timestamp": "2023-02-14T20:36:09Z",
  "files": [
    {
      "id": "4b6d37a3-a0ae-498a-a343-3ba2d6e075a0",
      "file_name": "sbomqs.cdx.json",
      "avg_score": 6.8,
      "scores": [
        {
          "category": "NTIA-minimum-elements",
          "feature": "Components have names",
          "score": 10.0,
          "max_score": 10.0,
          "description": "All components have names"
        }
      ]
    },
    {
      "id": "46b49389-f376-4100-b91f-025fcd0c810b",
      "file_name": "julia.spdx.json",
      "avg_score": 9.0,
      "scores": [
        {
          "category": "NTIA-minimum-elements",
          "feature": "Components have versions",
          "score": 10.0,
          "max_score": 10.0,
          "description": "All components have versions"
        }
      ]
    }
  ]
}
```

**Comment by surendrapathak** [comment]
1. Let's version the output to ensure we can incorporate new scoring criteria.
2. Let's include inferred SBOM spec, format, and version with the output (the consumption tool might branch for those).
3. I am confused about if we should break down DESC for the cases that include matching / non-matching scores (e.g., co 0/34) as well as cases that include lists (e.g., cpe:0, purl:0, total:34). I think for this release we go without breaking them down further but add them back with the "--debug" flag we discussed.

**Comment by riteshnoronha** [comment]
> 1. Let's version the output to ensure we can incorporate new scoring criteria.

This is a good idea, i believe over time these criteria would evolve. We should discuss some standard way to achieve 
this. 

> 2. Let's include inferred SBOM spec, format, and version with the output (the consumption tool might branch for those).

This can be done.  I will also add the sbomqs tool version. 

> 3. I am confused about if we should break down DESC for the cases that include matching / non-matching scores (e.g., co 0/34) as well as cases that include lists (e.g., cpe:0, purl:0, total:34). I think for this release we go without breaking them down further but add them back with the "--debug" flag we discussed.

Sounds good. 

> 4. We also need a response node for "Error"
Good idea.

**Comment by riteshnoronha** [comment]
@surendrapathak modified output based on feedback 

```json
{
  "run_id": "392d6634-3eae-4105-b1e4-26d139b4cdf1",
  "timestamp": "2023-02-14T22:12:04Z",
  "creation_info": {
    "name": "sbomqs",
    "version": "0.0.5",
    "scoring_engine_version": 1
  },
  "files": [
    {
      "id": "9c375c7c-ce71-49fe-8d3a-5f5914b5036d",
      "file_name": "sbomqs.cdx.json",
      "spec": "cyclonedx",
      "spec_version": "1.4",
      "file_format": "json",
      "avg_score": 6.8,
      "scores": [
        {
          "category": "NTIA-minimum-elements",
          "feature": "Components have names",
          "score": 10.0,
          "max_score": 10.0,
          "description": "All components have names"
        }
      ]
    },
    {
      "id": "9a4ce484-3e53-4a2e-b102-9a9d314de10e",
      "file_name": "julia.spdx.json",
      "avg_score": 9.0,
      "spec": "spdx",
      "spec_version": "2.2",
      "file_format": "json",
      "scores": [
        {
          "category": "NTIA-minimum-elements",
          "feature": "Components have names",
          "score": 10.0,
          "max_score": 10.0,
          "description": "All components have names"
        }
      ]
    }
  ]
}
```

**Comment by surendrapathak** [comment]
I think adding the information about the tool generating SBOM, and its version will be helpful for the consumer (for the cases where any generation tool mistakenly ships with a broken feature but was put through the pipelines and ended up generating SBOMs in production).

**Comment by riteshnoronha** [comment]
@surendrapathak thanks for the feedback. based on all our discussions below is the finalized output json 

- Removed ID from files, as it served no current purpose. Once we implement history, we can add it. 
- Add sbom tool creation info.
- Renamed scoring engine version and changed value to string

Will work on implementing this next week. 

```json
{
    "run_id": "392d6634-3eae-4105-b1e4-26d139b4cdf1",
    "timestamp": "2023-02-14T22:12:04Z",
    "creation_info": {
        "name": "syft",
        "version": "0.37.0"
    },
    "scoring_tool_info": {
      "name": "sbomqs",
      "version": "0.0.5",
      "scoring_version": "1"
    },
    "files": [
      {
        "file_name": "sbomqs.cdx.json",
        "spec": "cyclonedx",
        "spec_version": "1.4",
        "file_format": "json",
        "avg_score": 6.8,
        "scores": [
          {
            "category": "NTIA-minimum-elements",
            "feature": "Components have names",
            "score": 10.0,
            "max_score": 10.0,
            "description": "All components have names"
          }
        ]
      },
      {
        "file_name": "julia.spdx.json",
        "avg_score": 9.0,
        "spec": "spdx",
        "spec_version": "2.2",
        "file_format": "json",
        "scores": [
          {
            "category": "NTIA-minimum-elements",
            "feature": "Components have names",
            "score": 10.0,
            "max_score": 10.0,
            "description": "All components have names"
          }
        ]
      }
    ]
  }
```

**Comment by riteshnoronha** [comment]
@jspeed-meyers this feature has been released with v0.07 of the tool. If you do end up using it, would love any feedback.

**Comment by jspeed-meyers** [comment]
Thanks, @riteshnoronha! Will do.
